librz/core/cmd.c

```diff
@@ -4760,55 +4764,131 @@ DEFINE_HANDLE_TS_FCN_AND_SYMBOL(redirect_command) {
 	return res;
 }
 
+typedef struct _search_help {
+	bool color;
+	PJ* pj;
+} RzHelpSearch;
+
+static bool do_help_manual(RzCmd *cmd, const char *cmdname, bool color) {
+	if (!cmdname) {
+		return false;
+	}
+	char *cmdname_help = NULL;
+	char *help_msg = NULL;
+	RzCmdParsedArgs *help_pra = NULL;
+
+	cmdname_help = rz_str_newf("%s?", cmdname);
+	if (!cmdname_help) {
+		goto help_rec_err;
+	}
+	help_pra = rz_cmd_parsed_args_newcmd(cmdname_help);
+	if (!help_pra) {
+		goto help_rec_err;
+	}
+	ut32 pflags = color ? RZ_CMD_HELP_PRINT_USE_COLORS : 0;
+	help_msg = rz_cmd_get_help(cmd, help_pra, pflags);
+	if (!help_msg) {
+		goto help_rec_err;
+	}
+	rz_cons_printf("%s", help_msg);
+
+help_rec_err:
+	free (help_msg);
+	rz_cmd_parsed_args_free(help_pra);
+	free (cmdname_help);
+	return true;
+}
+
+static bool help_search_cmd_desc_entry(RzCmd *cmd, RzCmdDesc* cd, const char *name, void *user) {
+	rz_return_val_if_fail(cd, false);
+	if (cd->type != RZ_CMD_DESC_TYPE_ARGV &&
+		cd->type != RZ_CMD_DESC_TYPE_ARGV_MODES) {
+		return true;
+	}
+	RzHelpSearch* hs = (RzHelpSearch*) user;
+	if (hs->pj) {
+		rz_cmd_get_help_json(cmd, cd, name, hs->pj);
+	} else if (!strcmp(cd->name, name)) {
+		do_help_manual(cmd, cd->name, hs->color);
+	}
+	return true;
+}
+
+RZ_IPI RzCmdStatus rz_cmd_help_search_handler(RzCore *core, int argc, const char **argv, RzOutputMode mode) {
+	bool is_interactive = rz_cons_is_interactive(); 
+	rz_cons_set_interactive(false); 
+	PJ* pj = NULL;
+	RzCmdStatus status = RZ_CMD_STATUS_OK;
+
+	RzCmdDesc *begin = argc == 2 ? rz_cmd_get_desc(core->rcmd, argv[1]) : NULL;
+	if (argc == 2 && !begin) {
+		return RZ_CMD_STATUS_WRONG_ARGS;
+	}
+
+	if (mode & RZ_OUTPUT_MODE_JSON) {
+		pj = pj_new();
+		if (!pj) {
+			status = RZ_CMD_STATUS_ERROR;
+			goto exit_status;
+		}
+		pj_o(pj);
+	}
+
+	RzHelpSearch hs = {
+		.color = core->print->flags & RZ_PRINT_FLAGS_COLOR,
+		.pj = pj,
+	};
+
+	rz_cmd_foreach_cmdname(core->rcmd, begin, help_search_cmd_desc_entry, &hs);
+
+	if (mode & RZ_OUTPUT_MODE_JSON) {
+		pj_end(pj);
+		rz_cons_printf("%s\n", pj_string(pj));
+		pj_free(pj);
+	}
+exit_status:
+	rz_cons_set_interactive(is_interactive); 
+	return status;
+}
+
 DEFINE_HANDLE_TS_FCN_AND_SYMBOL(help_command) {
 	size_t node_str_len = strlen(node_string);
-	if (node_str_len >= 2 && !strcmp(node_string + node_str_len - 2, "?*")) {
```

Comment:
this can still be used to suppport the format `<cmd>?*`

Reply:
no. that needs to go to `argv_modes` to be supported and tbh, i do not think is a good thing.

i highly prefer using `?* <prefix>`

Reply:
Do support this for now. Because when you do `pd?*` the tree-sitter grammar will recognize a `help_command` and it will land here. So for now add it here, it should not be too hard as you already have all the methods.

Reply:
yes true.